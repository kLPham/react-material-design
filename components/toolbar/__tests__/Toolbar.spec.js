import React from 'react';
import { shallow } from 'enzyme';
import Toolbar from '../index';
import ToolbarRow from '../toolbarRow';

describe('<Toolbar />', () => {
    it('should render a toolbar', () => {
        const toolbar = shallow(
            <Toolbar />,
    );
        expect(toolbar).toMatchSnapshot();
    });
    it('should render fixed', () => {
        const fixed = true;
        const toolbar = shallow(
            <Toolbar fixed={fixed} />,
      );
        expect(toolbar).toMatchSnapshot();
    });
    it('should render with a title', () => {
        const title = 'This is my Title';
        const toolbar = shallow(
            <Toolbar title={title} />,
      );
        expect(toolbar).toMatchSnapshot();
    });
    it('should render with multiple rows', () => {
        const title = 'This is my title';
        const icon = <i className="material-icons">cloud</i>;
        const toolbar = shallow(
            <Toolbar title={title}>
                <ToolbarRow centerElements={icon} />
            </Toolbar>,
      );
        expect(toolbar).toMatchSnapshot();
    });
});
