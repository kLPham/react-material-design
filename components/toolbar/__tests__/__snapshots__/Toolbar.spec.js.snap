// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> should render a toolbar 1`] = `
<header
  className="mdc-toolbar"
>
  <ToolbarRow />
</header>
`;

exports[`<Toolbar /> should render fixed 1`] = `
<header
  className="mdc-toolbar mdc-toolbar--fixed"
>
  <ToolbarRow />
</header>
`;

exports[`<Toolbar /> should render with a title 1`] = `
<header
  className="mdc-toolbar"
>
  <ToolbarRow
    title="This is my Title"
  />
</header>
`;

exports[`<Toolbar /> should render with multiple rows 1`] = `
<header
  className="mdc-toolbar"
>
  <ToolbarRow
    title="This is my title"
  />
  <ToolbarRow
    centerElements={
      <i
        className="material-icons"
      >
        cloud
      </i>
    }
  />
</header>
`;
